wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineServiceLinksInjector.java

```diff
@@ -118,23 +118,29 @@ public MachineDto injectLinks(MachineDto machine, ServiceContext serviceContext)
 
         injectTerminalLink(machine, serviceContext, links);
 
-        // add links to websocket channels
-        final Link machineChannelLink = createLink("GET",
-                                                   serviceContext.getBaseUriBuilder()
-                                                                 .path("ws")
-                                                                 .path(machine.getWorkspaceId())
-                                                                 .scheme("https".equals(getLogsUri.getScheme()) ? "wss" : "ws")
-                                                                 .build()
-                                                                 .toString(),
-                                                   null);
+        // add workspace channel links
+        final Link workspaceChannelLink = createLink("GET",
+                                                     serviceContext.getBaseUriBuilder()
+                                                                   .path("ws")
+                                                                   .path(machine.getWorkspaceId())
+                                                                   .scheme("https".equals(getLogsUri.getScheme()) ? "wss" : "ws")
+                                                                   .build()
+                                                                   .toString(),
+                                                     null);
         final LinkParameter channelParameter = newDto(LinkParameter.class).withName("channel")
                                                                           .withRequired(true);
 
-        injectMachineChannelsLinks(machine.getConfig(),
-                                   machine.getWorkspaceId(),
-                                   machine.getEnvName(),
-                                   machineChannelLink,
-                                   channelParameter);
+        links.add(cloneDto(workspaceChannelLink).withRel(GET_ENVIRONMENT_OUTPUT_CHANNEL)
+                                                .withParameters(singletonList(cloneDto(channelParameter)
+                                                                                      .withDefaultValue("workspace:" +
+                                                                                                        machine.getWorkspaceId() +
+                                                                                                        ":environment_output"))));
```

Comment:
"environment_output" could be a constant ? as it's used at several places
or a constant "workspace:%s:environment_output" then we only use String format with selected workspaceId